### PR TITLE
Logging/get_data dynamic endpoint

### PR DESCRIPTION
**Pull request by mattestrada01**
function calls/initializations/successes sent to logger.debug
database info sent to logger.info
error info sent to logger.error

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -49,32 +58,36 @@ async def get_garage_data():
...
-        print(f"Current time: {current_time}")
-        if current_time.hour >= 8 and current_time.hour < 14:
+        logger.info(f"Current time: {current_time}")
+        if current_time.hour >= 8 and current_time.hour < 18:
             try:
                 # Between 8am-2pm, call endpoint
                 asyncio.run(get_garage_data())  
             except Exception as e:
-                print(f"An error occurred: {e}")
+                logger.error(f"An error occurred: {e}")
```
Comment (evanugarte) [inline server.py]: you can just do 
```py
except Exception:
    logger.exception("unable to get garage data")
```

and the exception will be included in your log
https://stackoverflow.com/a/5191885

File: server.py [hunk 2]
```diff
@@ -49,32 +58,36 @@ async def get_garage_data():
...
         else:
-            print("Stopping data retrieval as it's past 2:00 PM PST.")
+            logger.debug("Stopping data retrieval as it's past 2:00 PM PST.")
```
Comment (evanugarte) [inline server.py]: might what to log this as info, up to you
```suggestion
            logger.info("Stopping data retrieval as it's past 2:00 PM PST.")
```

File: sqlhelper.py [hunk 3]
```diff
@@ -13,27 +16,27 @@ def maybe_create_table(dbfile: str, garage_names):
...
     # delete_garage_data()
-    print(f"{garage}")
+    logger.info(f"{garage}")
     try:
         query = f"INSERT INTO {garage} (garage_fullness, time) VALUES (?, ?)"
         c.execute(query, [fullness, timestamp])
         conn.commit()
     except Exception as e:
-        print(e)
+        logger.error(e)
```
Comment (evanugarte) [inline sqlhelper.py]: see exception comment

**Comment by mattestrada01** [comment]
updated branch to include changes

File: sqlhelper.py [hunk 4]
```diff
@@ -43,21 +46,21 @@ def get_garage_data(dbfile: str, garage, time=None):
...
     c = conn.cursor()
     time_threshold = (datetime.now() - timedelta(weeks=2)).strftime('%Y-%m-%d %H:%M:%S')
-    print("TIME:",time_threshold)
+    logger.debug(f"TIME: {time_threshold}")
     try:
         query = f"DELETE FROM {garage} WHERE time < ?"
         c.execute(query, (time_threshold,))
         conn.commit()
-        print("Old data deleted")
+        logger.debug("Old data deleted")
```
Comment (pnavab) [inline sqlhelper.py:61]: what if we add a count of how many rows were deleted and add that in the debug

File: sqlhelper.py [hunk 5]
```diff
@@ -43,21 +46,21 @@ def get_garage_data(dbfile: str, garage, time=None):
...
     except Exception as e:
-        print(e)
+        logger.error(e)
```
Comment (pnavab) [inline sqlhelper.py:63]: logger.exception("error deleting data")

**Review by pnavab — APPROVED** [review]
few comments but approved

**Pull request closed without merging**